ironfish/src/indexers/minedBlocksIndexer.ts

```diff
@@ -0,0 +1,293 @@
+          for (const hash of hashes) {
+            await this.minedBlocks.del(hash, tx)
+          }
+        }
+      })
+    })
+  }
+
+  async open(
+    options: { upgrade?: boolean; load?: boolean } = { upgrade: true, load: true },
+  ): Promise<void> {
+    if (this.isOpen) {
+      return
+    }
+
+    this.isOpen = true
+    await this.openDB(options)
+
+    if (options.load) {
+      await this.load()
+    }
+  }
+
+  async close(): Promise<void> {
+    if (!this.isOpen) {
+      return
+    }
+
+    this.isOpen = false
+    await this.closeDB()
+  }
+
+  async openDB(options: { upgrade?: boolean } = { upgrade: true }): Promise<void> {
+    await this.files.mkdir(this.location, { recursive: true })
+    await this.database.open()
+
+    if (options.upgrade) {
+      await this.database.upgrade(DATABASE_VERSION)
+    }
+  }
+
+  async closeDB(): Promise<void> {
+    await this.database.close()
+  }
+
+  async load(): Promise<void> {
+    const headHash = await this.meta.get('headHash')
+    this.chainProcessor.hash = headHash ? Buffer.from(headHash, 'hex') : null
+  }
+
+  start(): void {
+    if (this.isStarted) {
+      return
+    }
+    this.isStarted = true
+
+    void this.eventLoop()
+  }
+
+  async stop(): Promise<void> {
+    if (!this.isStarted) {
+      return
+    }
+    this.isStarted = false
+
+    if (this.eventLoopTimeout) {
+      clearTimeout(this.eventLoopTimeout)
+    }
+
+    if (this.database.isOpen) {
+      await this.updateHeadHash(this.chainProcessor.hash)
+    }
+  }
+
+  async eventLoop(): Promise<void> {
+    if (!this.isStarted) {
+      return
+    }
+
+    await this.updateHead()
+
+    if (this.isStarted) {
+      this.eventLoopTimeout = setTimeout(() => void this.eventLoop(), 1000)
+    }
+  }
+
+  async updateHead(): Promise<void> {
+    const { hashChanged } = await this.chainProcessor.update()
+
+    if (hashChanged) {
+      this.logger.debug(
+        `Updated MinedBlocksIndexer Head: ${String(this.chainProcessor.hash?.toString('hex'))}`,
+      )
+    }
+  }
+
+  async updateHeadHash(headHash: Buffer | null, tx?: IDatabaseTransaction): Promise<void> {
+    const hashString = headHash && headHash.toString('hex')
+    await this.meta.put('headHash', hashString, tx)
+  }
+
+  async *getMinedBlocks({
+    scanForks,
+    start,
+    stop,
+  }: {
+    scanForks?: boolean
+    start?: number
+    stop?: number
+  }): AsyncGenerator<MinedBlock, void, unknown> {
+    const { start: begin, stop: end } = BlockchainUtils.getBlockRange(this.chain, {
+      start,
+      stop,
+    })
+
+    let minedBlocks = (await this.minedBlocks.getAllValues())
+      .filter((block) => {
+        if (begin <= block.sequence && block.sequence <= end) {
+          return block
+        }
+      })
+      .sort((a, b) => a.sequence - b.sequence)
+
+    if (!scanForks) {
+      minedBlocks = minedBlocks.filter((block) => block.main === true)
+    }
+
+    for (const block of minedBlocks) {
+      yield block
+    }
+  }
```

Comment:
We should make this actually able to iterate the results lazily, but to do that we need to avoid accumulating all the results. We can do that with an index by sequence.